docs/source/notes/resumption.rst

```diff
@@ -0,0 +1,134 @@
+|:play_or_pause_button:| Auto Resumption
+========================================
+
+Resuming from checkpoints is commonly used to recover from hardware failures (e.g. spot instances in the cloud being terminiated), loss spikes for large language models, or other unforseen errors. Our trainer supports resuming from checkpoints with the ``load_path`` argument (see :doc:`/trainer/checkpointing` for more details).
+
+.. testsetup::
+
+    import os
+    import shutil
+
+    from composer import Trainer
+
+    trainer = Trainer(
+        model=model,
+        train_dataloader=train_dataloader,
+        max_duration="1ep",
+        save_filename="ep{epoch}.pt",
+        save_folder="./path/to/checkpoints",
+        save_overwrite=True,
+        save_interval="1ep",  # Save checkpoints every epoch
+    )
+    trainer.fit()
+
+    assert os.path.exists("./path/to/checkpoints/ep1.pt")
+
+    if not os.path.exists("./path/to/checkpoints/ep25.pt"):
```

Comment:
Curious why the copy here?

Reply:
The reason is that for fast tests, we don't actually train for 25 epochs, so want to pretend the ep1 checkpoint is the ep25 checkpoint.

This code is actually copied from `resumption.rst`, and based on code in this PR: https://github.com/mosaicml/composer/commit/d025644e5786a84ac7edebbc6284813df1503c98.